docs/src/main/asciidoc/security-openid-connect-dev-services.adoc

```diff
@@ -0,0 +1,139 @@
+
+image::dev-ui-keycloak-password-grant-page.png[alt=Dev UI OpenId Connect Keycloak Page - Password Grant,role="center"]
+
+Set a registered user name, a relative service endpoint path, click on `Test Service` and you will a status code displayed in the `Results` field, such as `200`, `403`, `401` or `404`.
+
+You will see in the console something similar to:
+
+[source,shell]
+----
+2021-06-04 14:45:50,939 INFO  [io.qua.dev.key.run.KeycloakDevServicesRecorder] (DEV Console action) Getting token from 'http://localhost:32788/auth' for user 'bob' in realm 'quarkus' using client id 'quarkus-app'
+2021-06-04 14:45:51,055 INFO  [io.qua.dev.key.run.KeycloakDevServicesRecorder] (DEV Console action) Test token: eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICI1REl4VkFlZmhfY0lpNUZRRHVPOHN4QjlnaVVVMmlTMGtVUVl3Z2dFakZjIn0...
+2021-06-04 14:45:51,056 INFO  [io.qua.dev.key.run.KeycloakDevServicesRecorder] (DEV Console action) Sending token to 'http://localhost:8080/api/admin'
+2021-06-04 14:45:51,068 INFO  [io.qua.dev.key.run.KeycloakDevServicesRecorder] (DEV Console action) Result: 403
+----
+
+A token is acquired from keycloak using a `password` grant and is sent to the service endpoint.
+
+If you set `quarkus.keycloak.devservices.grant.type=client` then a `client_credentials` grant will be used to acquite a token, with the page showing no `User` field in this case.
+
+You can also sign in as an `admin:admin` directly to the launched Keycloak container, update the realm properties and export them for them to be reused after a restart.
+
+==== Developing OpenId Connect Web App Applications
+
+If you develop a link:security-openid-connect-web-authentication[Quarkus OIDC web-app application] then you should set `quarkus.oidc.application-type=web-app` in `application.properties` before starting an application.
+
+You will see a screen like this one:
+
+image::dev-ui-keycloak-sign-in.png[alt=Dev UI OpenId Connect Keycloak Sign In,role="center"]
+
+Set a relative service endpoint path, click on `Sign In To Service` and you will be redirected to Keycloak to enter a user name and password in a new browser tab.
+
+You can also sign in as an `admin:admin` directly to the launched Keycloak container, update the realm properties and export them for them to be reused after a restart.
+
+==== Keycloak Initialization
+
+You do not need to configure `quarkus-devservices-keycloak` or `quarkus-oidc` to start developing your Quarkus `OIDC` applications with the only exception is that `quarkus.oidc.application-type=web-app` has to be set in `application.properties` to give the `Keycloak` page a hint it needs to show an option to `Sign In To Service`.
+
+By default, the `quarkus`, `quarkus-app` client with a `secret` password, `alice` and `bob` users (with the passwords matching the names), and `user` and `admin` roles are created, with `alice` given both `admin` and `user` roles and `bob` - the `user` role. This should be enough to get started.
```

Comment:
I would just go 'user' and 'admin' instead of Alice and Bob. A lot of devs won't have studied crypto and won't get the reference.

Reply:
@stuartwdouglas We already use `alice` in all the quickstarts, tests (wiremock test resource manager creates it as well), docs. It is just a default user name we used with Pedro, without any particular references to the security scenarios. I'm OK with changing it to `user`/`admin` but we should probably do it separately to make it consistent across it all (quickstarts/docs/tests).